org.eclipse.scout.rt.dataobject/src/main/java/org/eclipse/scout/rt/dataobject/id/IdCodec.java

```diff
@@ -197,7 +205,10 @@ protected IId fromQualifiedInternal(String qualifiedId, boolean lenient) {
     String[] tmp = qualifiedId.split(":", 2); // split into at most two parts
     if (tmp.length < 2) { // no ":" found
       if (lenient) {
+        //noinspection deprecation
         return null;
+        // FIXME PBZ: To Discuss during code review: retain this case as UnknownId
```

Comment:
String to DO back to String.
In case we have a "foo" in the qualified case (DoValue<IId>), then the resulting string doesn't contain "foo" anymore, correct? With that, we recommend to add support here too.

Reply:
Done, Added rounttrip tests for both cases.